Neos.ContentRepository.Core/Classes/CommandHandler/CommandHookInterface.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\CommandHandler;
+
+use Neos\ContentRepository\Core\EventStore\EventsToPublish;
+
+/**
+ * TODO docs
+ *
+ * @api
+ */
+interface CommandHookInterface
+{
+    public function beforeHandle(CommandInterface $command): CommandInterface;
```

Comment:
to be discussed: should command hooks be allowed to return a _collection_ of commands?

Reply:
I decided against this idea because it creates a can of worms:
If hook 1 turns cmd1 into [cmd2,cmd3] what is hook2 supposed to process. and if hook2 adds yet another command, the first hook should be triggered again etc.